src/crawlers/browser_crawler.js

```diff
@@ -0,0 +1,523 @@
+            persistCookiesPerSession = true,
+            useSessionPool = true,
+            sessionPoolOptions,
+            proxyConfiguration,
+            browserPoolOptions,
+            preNavigationHooks = [],
+            postNavigationHooks = [],
+            ...basicCrawlerOptions
+        } = options;
+
+        if (!useSessionPool && persistCookiesPerSession) {
+            // @TODO: Maybe we could also automatically set persistCookiesPerSession to false when useSessionPool is false and log warning
+            throw new Error('You cannot use "persistCookiesPerSession" without "useSessionPool" set to true.');
+        }
+
+        super({
+            ...basicCrawlerOptions,
+            useSessionPool: false,
+            handleRequestFunction: (...args) => this._handleRequestFunction(...args),
+            handleRequestTimeoutSecs: handlePageTimeoutSecs * BASIC_CRAWLER_TIMEOUT_MULTIPLIER,
+        });
+
+        this.handlePageFunction = handlePageFunction;
+        this.handlePageTimeoutSecs = handlePageTimeoutSecs;
+        this.handlePageTimeoutMillis = this.handlePageTimeoutSecs * 1000;
+
+        this.gotoFunction = gotoFunction;
+        this.gotoOptions = {};
+
+        this.persistCookiesPerSession = persistCookiesPerSession;
+        this.proxyConfiguration = proxyConfiguration;
+
+        this.preNavigationHooks = preNavigationHooks;
+        this.postNavigationHooks = postNavigationHooks;
+
+        if (useSessionPool) {
+            this.sessionPool = new SessionPool({
+                ...sessionPoolOptions,
+                log: this.log,
+            });
+
+            // Assuming there are not more thant 20 browsers running at once;
+            this.sessionPool.setMaxListeners(20);
+        }
+
+        const { preLaunchHooks = [], postLaunchHooks = [], ...rest } = browserPoolOptions;
+        this.browserPool = new BrowserPool({
+            ...rest,
+            preLaunchHooks: [
+                this._extendLaunchContext.bind(this),
+                ...preLaunchHooks,
+            ],
+            postLaunchHooks: [
+                this._maybeAddSessionRetiredListener.bind(this),
+                ...postLaunchHooks,
+            ],
+        });
+    }
+
+    /**
+     * Wrapper around handlePageFunction that opens and closes pages etc.
+     *
+     * @param {Object} crawlingContext
+     * @param {Request} crawlingContext.request
+     * @param {BrowserController} crawlingContext.browserController
+     * @param {Session} [crawlingContext.session]
+     * @private
+     */
+    async _handleRequestFunction(crawlingContext) {
+        const { id } = crawlingContext;
+        const page = await this.browserPool.newPage({ id });
+        this._enhanceCrawlingContextWithPageInfo(crawlingContext, page);
+
+        const { request, session } = crawlingContext;
+
+        if (this.persistCookiesPerSession) {
+            const cookies = crawlingContext.session.getPuppeteerCookies(request.url);
```

Comment:
getPuppeteerCookies will work for Playwright too?

Reply:
Yes